### PR TITLE
chore: small improvements from `8a60ea2`

### PR DESCRIPTION
**Pull request by 0xteddybear**
- [x] lower hanging fruit #113
    - [x] empty-commitment
    - [x] viewlock on isValidSignature
    - [x] safeERC20 tests -- do collections break?
    - [x] _viewlock_ natspec
    - [x] link in docs
    - [x] clarify docs: there are two factories now
    - [x] 5 minutes
    - [x] unused ibfactory
    - [x] remove commitment method
    - [x] deploy amounts
    - [x] dont track smock/mockbcowpool
    - [x] internal_newbpool
    
    I'd drop the commit using SafeERC20 on bfactory because we know the token being transferred will return a boolean, since they're `BToken`s, deployed by the factory, and it has a non-zero gas increase

### CODE REVIEW

File: test/unit/BFactory.t.sol [hunk 1]
```diff
@@ -207,24 +208,44 @@ contract BFactory_Unit_Collect is BFactoryTest {
...
 
     vm.mockCall(_lpToken, abi.encodeWithSelector(IERC20.balanceOf.selector, address(bFactory)), abi.encode(_toCollect));
     vm.mockCall(_lpToken, abi.encodeWithSelector(IERC20.transfer.selector, owner, _toCollect), abi.encode(false));
 
-    vm.expectRevert(IBFactory.BFactory_ERC20TransferFailed.selector);
+    vm.expectRevert(abi.encodeWithSelector(SafeERC20.SafeERC20FailedOperation.selector, _lpToken));
     vm.prank(owner);
     bFactory.collect(IBPool(_lpToken));
   }
 }
```
Comment (0xAustrian) [inline test/unit/BFactory.t.sol:240]: If we are going to use SafeERC20, I would just remove this test.

File: src/contracts/BFactory.sol [hunk 2]
```diff
@@ -43,10 +44,7 @@ contract BFactory is IBFactory {
       revert BFactory_NotBLabs();
     }
     uint256 collected = pool.balanceOf(address(this));
-    bool xfer = pool.transfer(_blabs, collected);
-    if (!xfer) {
-      revert BFactory_ERC20TransferFailed();
-    }
+    SafeERC20.safeTransfer(pool, _blabs, collected);
```
Comment (0xAustrian) [inline src/contracts/BFactory.sol:47]: Should we use SafeERC20 in `_pullUnderlying` and `_pushUnderlying` (bPool) ?

Comment (0xteddybear) [reply src/contracts/BFactory.sol:47]: ...yes, it makes way more sense than here

File: src/contracts/BPool.sol [hunk 3]
```diff
@@ -42,7 +45,7 @@ contract BPool is BToken, BMath, IBPool {
     _setLock(_MUTEX_FREE);
   }
 
-  /// @dev Prevents reentrancy in view functions
+  /// @dev Throws an error when the reentrancy mutex is taken. Doesn't modify it.
```
Comment (wei3erHase) [inline src/contracts/BPool.sol:48]: 👍

File: test/unit/BCoWFactory.t.sol [hunk 4]
```diff
@@ -23,11 +30,10 @@ abstract contract BCoWFactoryTest is Base {
...
 
   function _bPoolBytecode() internal virtual override returns (bytes memory _bytecode) {
-    vm.skip(true);
-
     // NOTE: "runtimeCode" is not available for contracts containing immutable variables.
-    // return type(BCoWPool).runtimeCode;
-    return _bytecode;
+    // so we the easiest way to know the bytecode is to deploy it with the same
+    // parameters the factory would
+    return address(new BCoWPool(solutionSettler, appData)).code;
```
Comment (wei3erHase) [inline test/unit/BCoWFactory.t.sol:36]: god. 👏

File: test/unit/BFactory.t.sol (hunk 1 above)
Comment (wei3erHase) [reply test/unit/BFactory.t.sol:240]: not to bad to keep them imo

**Comment by wei3erHase** [comment]
Closes BAL-124, BAL-126

**Review by wei3erHase — APPROVED** [review]
amazing job here pal 👏

**Pull request merged**